src/radio.rs

```diff
@@ -446,64 +483,69 @@ impl Phy {
 use nrf52840_hal::pac::interrupt;
 #[interrupt]
 fn RADIO() {
-    irq_handler();
+    // Can be optimized with get_unchecked for release build.
+    irq_handler(Phy::get_global_instance());
```

Comment:
When correct ISR abstraction layer is implemented, I guess `Phy` structure during initialization could register the ISR providing `&self.isr_data` so that proper top-level ISR would call `irq_handler(registered_isr_data)`.

Temporarily solution like you suggested seems fine, but instead of passing whole `Phy` as a parameter it should be enough to use `IsrData`.